src/main.ts

```diff
@@ -1,10 +1,13 @@
 import computeWGSL from "./compute.wgsl?raw";
 import vertWGSL from "./vert.wgsl?raw";
 import fragWGSL from "./frag.wgsl?raw";
+import tgpu, { TgpuBuffer, Storage } from "typegpu";
+import { arrayOf, TgpuArray, U32, u32, vec2u } from "typegpu/data";
+import { layout0 as bindGroupLayoutCompute } from "./definitions/compute";
+import { layout0 as bindGroupLayoutRender } from "./definitions/vert";
 
 const canvas = document.querySelector("canvas") as HTMLCanvasElement;
-const adapter = await navigator.gpu?.requestAdapter();
-const device = await adapter?.requestDevice();
+const root = await tgpu.init();
```

Comment:
A small suggestion, since a lot of the APIs still require a device, this makes the code a bit simpler.
```suggestion
const root = await tgpu.init();
const device = root.device;
```